config/config.go

```diff
@@ -91,13 +91,14 @@ type GitConfig struct {
 	GitURLInsteadOf string `yaml:"url_instead_of"`
 
 	// these are used for git vendors (e.g. github, gitlab)
-	Pattern    string `yaml:"pattern"    help:"A regular expression defined to match git URLs, defaults to the regex: <site>/([^/]+)/([^/]+). For example if the site is github.com, then the default pattern will match github.com/<user>/<repo>."`
-	Substitute string `yaml:"substitute" help:"If specified, a regular expression substitution will be preformed to determine which URL is cloned by git. Values like $1, $2, ... will be replaced with matched subgroup data. If no substitute is given, a URL will be created based on the requested SSH authentication mode."`
-	Suffix     string `yaml:"suffix"     help:"The git repository suffix, like .git."`                                       // .git
-	Auth       string `yaml:"auth"       help:"What authentication method do you use? Valid options are: http, https, ssh."` // http, https, ssh
-	User       string `yaml:"user"       help:"The https username to use when auth is set to https. This setting is ignored when auth is ssh."`
-	Password   string `yaml:"password"   help:"The https password to use when auth is set to https. This setting is ignored when auth is ssh."`
-	KeyScan    string `yaml:"serverkey"  help:"SSH fingerprints, like you would add in your known hosts file, or get from ssh-keyscan."`
+	Pattern           string `yaml:"pattern"                      help:"A regular expression defined to match git URLs, defaults to the regex: <site>/([^/]+)/([^/]+). For example if the site is github.com, then the default pattern will match github.com/<user>/<repo>."`
+	Substitute        string `yaml:"substitute"                   help:"If specified, a regular expression substitution will be preformed to determine which URL is cloned by git. Values like $1, $2, ... will be replaced with matched subgroup data. If no substitute is given, a URL will be created based on the requested SSH authentication mode."`
+	Suffix            string `yaml:"suffix"                       help:"The git repository suffix, like .git."`                                       // .git
+	Auth              string `yaml:"auth"                         help:"What authentication method do you use? Valid options are: http, https, ssh."` // http, https, ssh
+	User              string `yaml:"user"                         help:"The username to use when auth is set to git or https."`
+	Password          string `yaml:"password"                     help:"The https password to use when auth is set to https. This setting is ignored when auth is ssh."`
+	KeyScan           string `yaml:"serverkey"                    help:"SSH fingerprints, like you would add in your known hosts file, or get from ssh-keyscan."`
+	AllowEmptyKeyScan bool   `yaml:"insecure_allow_unknown_hosts" help:"Allow ssh access to hosts with unknown server keys (e.g. no entries in known_hosts)."`
```

Comment:
Should the Go field should match the yaml field name?

Reply:
possibly? do you think we should include the `Insecure` prefix? i.e. `InsecureAllowUnknownHosts` or just `AllowUnknownHosts`?

I don't have any strong preference of renaming.

Reply:
Not sure what the best name is to be honest, but the name of the Go field (AllowEmptyKeyScan) and the yaml field (insecure_allow_unknown_hosts) should be consistent at least IMO. I guess it would be useful to prefix with "Insecure" if we really thought it was and people shouldn't use it :thinking:

Reply:
I renamed it to `InsecureAllowUnknownHosts`, and additionally changed `KeyScan` to `ServerKey`.